src/main/java/myjournal/Parser.java

```diff
@@ -0,0 +1,292 @@
+        }
+        tasks.getTask(index).setState(true);
+        return tasks.getTask(index);
+    }
+
+    /**
+     * Parses user's input for the command "list".
+     *
+     * @param tasks The list of tasks.
+     */
+    public static String parseList(TaskList tasks) {
+        if (tasks.getSize() == 0) {
+            return "You have no task!";
+        } else {
+            String s = "Here are your tasks:\n";
+            for (int i = 0; i < tasks.getSize(); i++) {
+                s = s + (i + 1) + "." + tasks.getTask(i) + "\n";
+            }
+            return s;
+        }
+    }
+
+    /**
+     * Parses user's input for the command "delete".
+     *
+     * @param line The user's input.
+     * @param tasks The list of tasks.
+     * @return The task to be deleted.
+     */
+    public static Task parseDelete(Scanner line, TaskList tasks) {
+        if (!line.hasNextInt()) {
+            throw new InvalidTaskNumberException("OOPS!!! Please specify the task "
+                    + "that needs to be deleted!");
+        }
+        int index = line.nextInt() - 1;
+        if (index >= tasks.getSize() || index < 0 || tasks.getTask(index) == null) {
+            throw new InvalidTaskNumberException("OOPS!!! Please enter a valid task number!");
+        }
+        Task temp = tasks.getTask(index);
+        tasks.deleteTask(index);
+        return temp;
+    }
+
+    /**
+     * Returns the time or date of the task.
+     *
+     * @param line The user's input.
+     * @return The String representation of the time or date of the task.
+     * @throws DateTimeParseException An exception thrown if the date or time is invalid.
+     */
+    public static String getTimeDate(Scanner line) throws DateTimeParseException {
+        String parsed = "";
+        while (line.hasNext()) {
+            String currWord = line.next();
+            if (isDate(currWord)) {
+                LocalDate date = LocalDate.parse(currWord);
+                String month = date.getMonth().getDisplayName(TextStyle.SHORT, Locale.ENGLISH);
+                String dateFormatted = date.getDayOfWeek().toString() + ", " + date.getDayOfMonth() + " " + month
+                        + " " + date.getYear();
+                parsed = parsed + " " + dateFormatted;
+            } else if (isTime(currWord)) {
+                LocalTime time = LocalTime.parse(currWord);
+                String beforeOrAfterNoon = time.getHour() >= 12 ? "pm" : "am";
+                int hour = time.getHour() == 0
+                        ? 12
+                        : time.getHour() >= 12
+                        ? time.getHour() - 12
+                        : time.getHour();
+                int min = time.getMinute();
+                String timeFormatted = (String.valueOf(hour).length() == 1 ? "0" + hour : hour)
+                        + ":" + (min < 10 ? "0" + min : min) + beforeOrAfterNoon;
+                parsed = parsed + " " + timeFormatted;
```

Comment:
Perhaps you can consider using [DateTimeFormatter](https://www.baeldung.com/java-datetimeformatter) here to simplify your code?  😊

Reply:
Okay I’ll try!! Thanks for the suggestion!:)